### PR TITLE
Add a document to explain how to install ROS and Gazebo together

### PR DESCRIPTION
**Pull request by j-rivero**
The PR provides a note in the default landing page and a document describing how should a user install ROS and Gazebo together. This is mostly a port of the current gazebo-classic: https://classic.gazebosim.org/tutorials?tut=ros_wrapper_versions.

The document provides an easy path for going with default Gazebo version in the ROS distributions and information about how to use other from source possible installations.

Depends on #338

### CODE REVIEW

File: get_started.md [hunk 1]
```diff
@@ -7,6 +7,9 @@ simulation using Gazebo.
 
 ## Step 1: Install
 
+***Note:*** If you are a [ROS](ros.org) user, please read first the tutorial about
+the [ROS/Gz installation](docs/ros_installation.md).
```
Comment (ahcorde) [inline get_started.md:11]:
```suggestion
the [ROS/Gz installation](./ros_installation.md).
```

File: ros_installation.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+This document provides an overview about the options to use different versions
+of ROS in combination with different versions of Gz. It is recommended to read
+it before installing the [ros_gz bridge](https://github.com/gazebosim/ros_gz).
+
+# Important! Simple analysis for a quick and correct decision
+
+If you are planning on using a specific version of ROS and don't have a reason
+to use a specific version of Gazebo, you should proceed with the default
+version of Gazebo available from the ROS repository when installing the ros_gz
```
Comment (ahcorde) [inline ros_installation.md:11]:
```suggestion
version of Gazebo available from the ROS repository when installing the `ros_gz`
```

File: ros_installation.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+bridge:
+
+```
+apt-get install ros-${ROS_DISTRO}-ros-gz
+
+```
```
Comment (ahcorde) [inline ros_installation.md:17]:
```suggestion
```bash
sudo apt-get install ros-${ROS_DISTRO}-ros-gz
```
```

File: ros_installation.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+This command is going to install the Gazebo libraries corresponding to the
+version detailed below in the section: "Using the default Gazebo version
+for a ROS distribution using binary installations". The rest of this
+document provides more information and other possible installations but
+at this point ROS and Gz are installed and ready.
+
+# Gazebo versions and ROS integration
+
+Gazebo is an independent project like boost, ogre or any other project used by
```
Comment (ahcorde) [inline ros_installation.md:27]:
```suggestion
Gazebo is an independent project like boost, Ogre or any other project used by
```

File: ros_installation.md [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+### Gz Garden
+
+Gz Garden can be used with ROS2 Humble and ROS2 Rolling, the [`ros_gz
+bridge`](https://github.com/gazebosim/ros_gz) supports the compilations from
+source.
+
+ * Folow [the instruction to install gz-garden](https://gazebosim.org/docs/garden/install_ubuntu#binary-installation-on-ubuntu)
+   from osrfoundation.org repository
+ * Follow the instructions to compile the ros_gz bridge from source in a colcon workspace
```
Comment (ahcorde) [inline ros_installation.md:124]:
```suggestion
 * Follow the instructions to compile the `ros_gz` bridge from source in a colcon workspace
```

File: ros_installation.md [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+   * [ROS2 Humble](https://github.com/gazebosim/ros_gz/tree/humble#from-source)
+     * Be sure of using `export GZ_VERSION=humble`
+   * [ROS2 Rolling](https://github.com/gazebosim/ros_gz/tree/ros2#from-source)
+     * Be sure of using `export GZ_VERSION=rolling`
```
Comment (ahcorde) [inline ros_installation.md:128]: This is wrong, maybe `ROS_VERSION`?

File: ros_installation.md [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+bridge`](https://github.com/gazebosim/ros_gz) supports the compilation from
+source.
+
+ * Folow [the instruction to install gz-edifice](https://gazebosim.org/docs/edifice/install_ubuntu#binary-installation-on-ubuntu)
+   from osrfoundation.org repository
+ * Follow the instructions to compile the ros_gz bridge from source in a colcon workspace
+   * [ROS2 Foxy](https://github.com/gazebosim/ros_gz/tree/foxy#from-source)
+     * Be sure of using `export GZ_VERSION=foxy`
+   * [ROS2 Rolling](https://github.com/gazebosim/ros_gz/tree/ros2#from-source)
+     * Be sure of using `export GZ_VERSION=rolling`
```
Comment (ahcorde) [inline ros_installation.md:155]: This is wrong GZ_VERSION should't be `rolling`

File: ros_installation.md [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+   * [ROS1 Noetic]()https://github.com/gazebosim/ros_gz/tree/noetic#from-source
+     * Be sure of using `export GZ_VERSION=edifice`
+
+## Using the latest Gz binary versions
+
+The Gz team usually backport and release new versions of each of the supported
+Gz releases and libraries (i.e: bumping gz-sim 7.0.0 to gz-sim 7.1.0). These
```
Comment (ahcorde) [inline ros_installation.md:162]:
```suggestion
Gz releases and libraries (i.e: bumping `gz-sim 7.0.0` to gz-sim `7.1.0`). These
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: ros_installation.md [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+version detailed below in the section: "Using the default Gazebo version
+for a ROS distribution using binary installations". The rest of this
+document provides more information and other possible installations but
+at this point ROS and Gz are installed and ready.
+
+# Gazebo versions and ROS integration
+
+Gazebo is an independent project like boost, ogre or any other project used by
+ROS. Usually, the latest version of Gazebo available at the beginning of every
+ROS release cycle (for example Gazebo Fortress for ROS2 Humble) is selected as
```
Comment (kscottz) [inline ros_installation.md:29]:
```suggestion
ROS release cycle (for example Gazebo Fortress for ROS 2 Humble) is selected as
```

File: ros_installation.md [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+versioning](http://semver.org/) philosophy for ABI stability, in which all
+versions that have the same major number (`gz-sim7_7.0.0`, `gz-sim7_7.1.0`,
+`gz-sim7_7.0.1`, ...) are binary compatible and thus interchangeable when using
+the same ROS distro.
+
+## Installing Gazebo
+
+### Gz simulator Ubuntu packages
+
+The easiest way of installing Gazebo is to use packages. There are two main
```
Comment (kscottz) [inline ros_installation.md:43]:
```suggestion
The easiest way of installing Gazebo is to use binary packages. There are two main
```

File: ros_installation.md [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+   * Gz Citadel
+   * Gz Edifice
+   * Gz Fortress
+   * Gz Garden
+
+This means that including the osrfoundation repository is not strictly needed
+to get the Gazebo binary package in Ubuntu. It can be installed from the ROS
+repository.
+
+## Using the default Gazebo version for a ROS distribution using binary installations
```
Comment (kscottz) [inline ros_installation.md:64]:
```suggestion
## Installing the default Gazebo version for a ROS distribution using binary installations
```

File: ros_installation.md [hunk 12]
```diff
@@ -0,0 +1,197 @@
...
+
+For the users that need to run a specific version of ROS and want to use all
```
Comment (kscottz) [inline ros_installation.md:66]:
```suggestion
For users that need to run a specific version of ROS and want to use all
```

File: ros_installation.md [hunk 13]
```diff
@@ -0,0 +1,197 @@
...
+the gazebo ROS related packages out-of-the-box, this is the recommended
+section:
+
+### ROS2 Humble and ROS2 Rolling
+ROS2 Humble and ROS2 Rolling support officially the Gazebo Fortress version.
```
Comment (kscottz) [inline ros_installation.md:71]:
```suggestion
ROS 2 Humble and ROS 2 Rolling support officially the Gazebo Fortress version.
```

File: ros_installation.md [hunk 14]
```diff
@@ -0,0 +1,197 @@
...
+For a fully-integrated ROS system, Fortress is the recommened version. The way
+to proceed is just to use the ROS repository and do ***not*** use the
```
Comment (kscottz) [inline ros_installation.md:73]: To clarify,  when you say "use" here you specifically mean "download and install binaries from" correct?

File: ros_installation.md [hunk 15]
```diff
@@ -0,0 +1,197 @@
...
+osrfoundation repository.
+
+ROS packages `ros-humble-ros-gz` and `ros-rolling-ros-gz` will be available
+pulling the necessary Gazebo Fortress dependencies.
```
Comment (kscottz) [inline ros_installation.md:77]: I think what you mean here is:
```suggestion
, these packages automatically install the necessary dependencies for Gazebo Fortress.
```

File: ros_installation.md [hunk 16]
```diff
@@ -0,0 +1,197 @@
...
+
+### ROS2 Galactic
```
Comment (kscottz) [inline ros_installation.md:79]:
```suggestion
### ROS 2 Galactic
```

File: ros_installation.md [hunk 17]
```diff
@@ -0,0 +1,197 @@
...
+ROS2 Galactic supports officially the Gazebo Edifice
```
Comment (kscottz) [inline ros_installation.md:80]:
```suggestion
ROS 2 Galactic officially supports Gazebo Edifice
```

File: ros_installation.md [hunk 18]
```diff
@@ -0,0 +1,197 @@
...
+version. For a fully-integrated ROS system, Edifice is the recommened
```
Comment (kscottz) [inline ros_installation.md:81]:
```suggestion
 For a fully-integrated ROS system, Edifice is the recommened
```

File: ros_installation.md [hunk 19]
```diff
@@ -0,0 +1,197 @@
...
+version. The way to proceed is just to use the ROS repository and do ***not***
```
Comment (kscottz) [inline ros_installation.md:82]: We have to be a bit clearer here.  I am not sure what use is supposed to mean.

File: ros_installation.md [hunk 20]
```diff
@@ -0,0 +1,197 @@
...
+use the osrfoundation repository.
+
+ROS packages `ros-galactic-ros-gz` will be available pulling the necessary
+Gazebo Edifice dependencies.
+
+### ROS2 Foxy
```
Comment (kscottz) [inline ros_installation.md:88]:
```suggestion
### ROS 2 Foxy
```

File: ros_installation.md [hunk 21]
```diff
@@ -0,0 +1,197 @@
...
+ROS2 Foxy supports officially the Gazebo Citadel
```
Comment (kscottz) [inline ros_installation.md:89]:
```suggestion
ROS 2 Foxy supports officially the Gazebo Citadel
```

File: ros_installation.md [hunk 22]
```diff
@@ -0,0 +1,197 @@
...
+version. For a fully-integrated ROS system, Citadel is the recommened
+version. The way to proceed is just to use the ROS repository and do ***not***
+use the osrfoundation repository.
+
+ROS packages `ros-foxy-ros-gz` will be available pulling the necessary
+Gazebo Citadel dependencies.
+
+### ROS1 Noetic
```
Comment (kscottz) [inline ros_installation.md:97]:
```suggestion
### ROS 1 Noetic
```

File: ros_installation.md [hunk 23]
```diff
@@ -0,0 +1,197 @@
...
+ROS1 Noetic supports officially the Gazebo Citadel
```
Comment (kscottz) [inline ros_installation.md:98]:
```suggestion
ROS Noetic officially supports Gazebo Citadel
```

File: ros_installation.md [hunk 24]
```diff
@@ -0,0 +1,197 @@
...
+   from osrfoundation.org repository
+ * Follow the instructions to compile the ros_gz bridge from source in a colcon workspace
+   * [ROS2 Humble](https://github.com/gazebosim/ros_gz/tree/humble#from-source)
+     * Be sure of using `export GZ_VERSION=humble`
+   * [ROS2 Rolling](https://github.com/gazebosim/ros_gz/tree/ros2#from-source)
+     * Be sure of using `export GZ_VERSION=rolling`
+
+### Gz Fortress
+
+Gz Fortress can be also used with ROS2 Galactic and ROS1 Noetic, the [`ros_gz
```
Comment (kscottz) [inline ros_installation.md:132]:
```suggestion
Gz Fortress can be also used with ROS 2 Galactic and ROS 1 Noetic, the [`ros_gz
```

File: ros_installation.md [hunk 25]
```diff
@@ -0,0 +1,197 @@
...
+
+### Gz Edifice
+Gz Edifice can be also used with ROS2 Foxy, ROS2 Rolling and ROS1 Noetic, the [`ros_gz
+bridge`](https://github.com/gazebosim/ros_gz) supports the compilation from
+source.
+
+ * Folow [the instruction to install gz-edifice](https://gazebosim.org/docs/edifice/install_ubuntu#binary-installation-on-ubuntu)
+   from osrfoundation.org repository
+ * Follow the instructions to compile the ros_gz bridge from source in a colcon workspace
+   * [ROS2 Foxy](https://github.com/gazebosim/ros_gz/tree/foxy#from-source)
```
Comment (kscottz) [inline ros_installation.md:152]:
```suggestion
   * [ROS 2 Foxy](https://github.com/gazebosim/ros_gz/tree/foxy#from-source)
```

File: ros_installation.md [hunk 26]
```diff
@@ -0,0 +1,197 @@
...
+     * Be sure of using `export GZ_VERSION=foxy`
+   * [ROS2 Rolling](https://github.com/gazebosim/ros_gz/tree/ros2#from-source)
```
Comment (kscottz) [inline ros_installation.md:154]:
```suggestion
   * [ROS 2 Rolling](https://github.com/gazebosim/ros_gz/tree/ros2#from-source)
```

**Review by kscottz — COMMENTED** [review]
I need to run, but I left a lot of fixes. I also need a couple of clarifications. This is way better than what we have but it probably needs a second editing pass (editing is hard as a pull request). If you want to merge this I can come back in next week and clean it up.

File: ros_installation.md (hunk 14 above)
Comment (j-rivero) [reply ros_installation.md:73]: "use the X repository" I meant ... setup keys and sources.lists for the X repository. So yes, at the end "download and install binaries from X repository".

File: ros_installation.md (hunk 6 above)
Comment (j-rivero) [reply ros_installation.md]: ouch, [4dd64b9](https://github.com/gazebosim/docs/pull/339/commits/4dd64b91e508fd7fc8e856d6a719ad2e65ed3b98) . Good catch.

File: ros_installation.md (hunk 7 above)
Comment (j-rivero) [reply ros_installation.md]: right, all them should be Edifice [b37e02e](https://github.com/gazebosim/docs/pull/339/commits/b37e02eddb80bc21d66b3cff4f3c9589f3812272)

File: ros_installation.md [hunk 27]
```diff
@@ -0,0 +1,196 @@
...
+For a fully-integrated ROS system, Fortress is the recommened version. The way
+to proceed is just to use the ROS repository and do ***not*** use the
+osrfoundation repository.
+
+ROS packages `ros-humble-ros-gz` and `ros-rolling-ros-gz` will be available
+, these packages automatically install the necessary dependencies for Gazebo Fortress.
+
+### ROS 2 Galactic
+ROS 2 Galactic officially supports Gazebo Edifice
+ For a fully-integrated ROS system, Edifice is the recommened
```
Comment (ahcorde) [inline ros_installation.md:80]:
```suggestion
For a fully-integrated ROS system, Edifice is the recommened
```

File: ros_installation.md [hunk 28]
```diff
@@ -0,0 +1,196 @@
...
+### ROS2 Humble and ROS2 Rolling
+ROS 2 Humble and ROS 2 Rolling support officially the Gazebo Fortress version.
+For a fully-integrated ROS system, Fortress is the recommened version. The way
+to proceed is just to use the ROS repository and do ***not*** use the
+osrfoundation repository.
+
+ROS packages `ros-humble-ros-gz` and `ros-rolling-ros-gz` will be available
+, these packages automatically install the necessary dependencies for Gazebo Fortress.
+
+### ROS 2 Galactic
```
Comment (ahcorde) [inline ros_installation.md:78]: End of life ?

```suggestion
### ROS 2 Galactic 
```

File: ros_installation.md [hunk 29]
```diff
@@ -0,0 +1,196 @@
...
+source.
+
+ * Folow [the instruction to install gz-edifice](https://gazebosim.org/docs/edifice/install_ubuntu#binary-installation-on-ubuntu)
+   from osrfoundation.org repository
+ * Follow the instructions to compile the ros_gz bridge from source in a colcon workspace
+   * [ROS 2 Foxy](https://github.com/gazebosim/ros_gz/tree/foxy#from-source)
+     * Be sure of using `export GZ_VERSION=edifice`
+   * [ROS 2 Rolling](https://github.com/gazebosim/ros_gz/tree/ros2#from-source)
+     * Be sure of using `export GZ_VERSION=edifice`
+   * [ROS1 Noetic]()https://github.com/gazebosim/ros_gz/tree/noetic#from-source
```
Comment (ahcorde) [inline ros_installation.md:155]:
```suggestion
   * [ROS1 Noetic](https://github.com/gazebosim/ros_gz/tree/noetic#from-source)
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

**Comment by j-rivero** [comment]
I've also include rosdep rules for Garden [4dff8d2](https://github.com/gazebosim/docs/pull/339/commits/4dff8d22faca6bb6066bc812075fe408807e60a8).

**Review by ahcorde — APPROVED** [review]

**Comment by j-rivero** [comment]
@azeey I think we can also merge this one to go in today's deployment. I have a pending clarification with @kscottz that might go into the next deployement after finding a good solution.

File: ros_installation.md [hunk 30]
```diff
@@ -0,0 +1,197 @@
+# Install Gazebo and ROS
+
+This document provides an overview about the options to use different versions
+of ROS in combination with different versions of Gz. It is recommended to read
```
Comment (azeey) [inline ros_installation.md:4]: nit: Gz -> Gazebo.
Here and below

File: ros_installation.md [hunk 31]
```diff
@@ -0,0 +1,197 @@
...
+Gazebo Citadel dependencies.
+
+## Using an specific Gazebo version with ROS 2
+***Warning!:*** Using this option, you won't be able to use any ROS Ubuntu package
+related to Gz libraries from ROS deb repository.
+
+Selecting a different release of Gz than the one officially supported in each
+ROS distribution will require that you manually compile the [`ros_gz
+bridge`](https://github.com/gazebosim/ros_gz) from source. Same applies for
+other ROS packages using Gz libraries that could also require code changes in
```
Comment (azeey) [inline ros_installation.md:112]: "changes in ..." is there something missing there?

File: ros_installation.md [hunk 32]
```diff
@@ -0,0 +1,197 @@
...
+
+### Gz Garden
+
+Gz Garden can be used with ROS 2 Humble and ROS 2 Rolling, the [`ros_gz
+bridge`](https://github.com/gazebosim/ros_gz) supports the compilations from
+source.
+
+ * Folow [the instruction to install gz-garden](https://gazebosim.org/docs/garden/install_ubuntu#binary-installation-on-ubuntu)
+   from osrfoundation.org repository
+ * Install [rosdep rules for Gazebo Garden](https://github.com/osrf/osrf-rosdep#installing-rosdep-rules-to-resolve-gazebo-garden-libraries)
```
Comment (azeey) [inline ros_installation.md:123]: wow, cool! I didn't know about this.

File: ros_installation.md [hunk 33]
```diff
@@ -0,0 +1,197 @@
...
+
+#### I want to use Gz Garden with ROS. Where are the packages?
+
+Unfortunately there are no Gz Garden packages for any ROS distribution. Please
+read how to compile it from source in this document under the section
+"Using an specific Gazebo version with ROS2" >> "Gz Garden"
+
+#### Where I can find the different features implemented on each Gazebo version?
+
+Some notes are regularly posted on the [Gazebo community
```
Comment (azeey) [inline ros_installation.md:187]: I think https://gazebosim.org/docs/garden/release-features would be a good link to put here.

**Review by azeey — APPROVED** [review]
Just minor comments. Very helpful stuff here. Thanks Jose!

File: ros_installation.md (hunk 32 above)
Comment (j-rivero) [reply ros_installation.md:123]: brand new, from yesterday. Should help users with Garden installations in ROS.

File: ros_installation.md (hunk 33 above)
Comment (j-rivero) [reply ros_installation.md:190]: Indeed, that is the best option. [572f672](https://github.com/gazebosim/docs/pull/339/commits/572f672e2208102c39713e1be07577407418d124)

File: ros_installation.md (hunk 31 above)
Comment (j-rivero) [reply ros_installation.md]: Probably I meant an "if" [a828de5](https://github.com/gazebosim/docs/pull/339/commits/a828de56d9c787b7c2ac52daea497a708453edef)

File: ros_installation.md (hunk 30 above)
Comment (j-rivero) [reply ros_installation.md]: replaced all them [03cd75f](https://github.com/gazebosim/docs/pull/339/commits/03cd75fdd4c1b2f6fa0d0d8895f1168e67233205)

**Comment by kscottz** [comment]
> @azeey I think we can also merge this one to go in today's deployment. I have a pending clarification with @kscottz that might go into the next deployment after finding a good solution.

Can you clarify a bit? Sorry to cut and run but I was headed out for a holiday weekend? Which comment in particular. Happy to do a second pass if it would help.

**Comment by j-rivero** [comment]
> > 
> 
> @j-rivero Can you clarify a bit? Sorry to cut and run but I was headed out for a holiday weekend? Which comment in particular. Happy to do a second pass if it would help.

No worries, this one https://github.com/gazebosim/docs/pull/339/files#r1107777671

**Pull request merged**